libraries/plugins/witness/witness_plugin.cpp

```diff
@@ -522,32 +522,35 @@ void witness_plugin::plugin_shutdown()
     return;
 }
 
-void witness_plugin::schedule_production_loop()
+void witness_plugin::schedule_production_loop(block_production_condition::block_production_condition_enum last_result)
 {
+    static const int64_t ONE_SECOND_MS = 1000000;
     // Schedule for the next second's tick regardless of chain state
     // If we would wait less than 50ms, wait for the whole second.
     fc::time_point fc_now = fc::time_point::now();
-    int64_t time_to_next_second = 1000000 - (fc_now.time_since_epoch().count() % 1000000);
+    int64_t time_to_next_second = ONE_SECOND_MS - (fc_now.time_since_epoch().count() % ONE_SECOND_MS);
     if (time_to_next_second < 50000) // we must sleep for at least 50ms
     {
-        time_to_next_second += 1000000;
+        time_to_next_second += ONE_SECOND_MS;
     }
 
     fc::time_point next_wakeup(fc_now + fc::microseconds(time_to_next_second));
 
+    dlog("last_result = ${r}", ("r", last_result));
```

Comment:
why don't you log in place where you do the call?

Reply:
It is helpful to understand. It is DEBUG log!

Reply:
I'm not against of DEBUG log! Put it in debug log in place where you received this value.